node/src/consensus/process_live_dispatcher.rs

```diff
@@ -44,12 +33,17 @@ impl ProcessLiveDispatcher {
             self.election_schedulers.activate(tx, &block.account());
         }
 
-        if let Some(websocket) = &self.websocket {
-            if websocket.any_subscriber(Topic::NewUnconfirmedBlock) {
-                websocket.broadcast(&new_block_arrived_message(block));
+        {
+            let callbacks = self.new_unconfirmed_block_observer.lock().unwrap();
```

Comment:
This will hold the lock until all observers are done and this brings the risk of deadlocks. The observers should be copied and the lock dropped before we call them